scripts/run-bundled-codegen.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+
+# Exit on all errors, undeclared variables and pipefailures.
+set -euo pipefail
+
+# Get the path to the script directory
+SCRIPT_DIR="$(dirname $0)"
+
+# Get the SHASUM of the tarball
+ZIP_FILE="${SCRIPT_DIR}/apollo.tar.gz"
+SHASUM="$(/usr/bin/shasum -a 256 ${ZIP_FILE})"
+SHASUM_FILE="${SCRIPT_DIR}/apollo/.shasum"
+APOLLO_DIR="${SCRIPT_DIR}"/apollo
+
+# Helper functions
+remove_existing_apollo() {
+  rm -r "${APOLLO_DIR}"
+}
+
+extract_cli() {
+  tar xzf "${SCRIPT_DIR}"/apollo.tar.gz -C "${SCRIPT_DIR}"
+  echo "${SHASUM}" | tee "${SHASUM_FILE}"
+}
+
+check_shasum() {
+  # Make sure the SHASUM matches the release for this version
+  EXPECTED_SHASUM="13febaa462e56679099d81502d530e16c3ddf1c6c2db06abe3822c0ef79fb9d2  ${ZIP_FILE}"
+
+  if [ "${SHASUM}" == "${EXPECTED_SHASUM}" ]; then
+    echo "Correct version of the CLI tarball is included, checking if it's already been extracted..."
+  else
+    echo "Error: The SHASUM of this zip file does not match the official released version from Apollo! This may present security issues. Terminating code generation."
```

Comment:
You might want to echo this to stderr instead of stdout. You can do so by redirecting the output by appending ` >&2` to the end of the line.

Reply:
Doesn't seem to have any outward effect when building with Xcode directly, but couldn't hurt for command line builds.

Reply:
Added